ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -485,6 +480,26 @@ private void clearLocksAndHB() {
     stopHeartbeat();
   }
 
+  private void cleanupDirForCTAS() {
+    if (HiveConf.getBoolVar(conf, HiveConf.ConfVars.TXN_CTAS_X_LOCK)) {
+      if (destinationTable != null) {
+        try {
+          CompactionRequest rqst = new CompactionRequest(
+                  destinationTable.getDbName(), destinationTable.getTableName(), CompactionType.MAJOR);
+          rqst.setRunas(TxnUtils.findUserToRunAs(destinationTable.getSd().getLocation(),
+                  destinationTable.getTTable(), conf));
+
+          rqst.putToProperties("location", destinationTable.getSd().getLocation());
+          rqst.putToProperties("ifPurge", Boolean.toString(true));
```

Comment:
would be good if we move "ifPurge" under constants as well

Reply:
Create a constant called `IF_PURGE` . Done.